solidserver/data_source_dns_zone.go

```diff
@@ -0,0 +1,115 @@
+package solidserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"log"
+	"net/url"
+)
+
+
+func dataSourcednszone() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourcednszoneRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "The Domain Name served by the zone.",
+				Required:    true,
+			},
+			"type": {
+				Type:		schema.TypeString,
+				Description: "The Type of the DNS zone.",
+				Optional:    true,
+				Default:      "master",
+			},
+			"dns_name": {
+				Type:		schema.TypeString,
+				Description: "The DNS name",
+				Computed:    true,
+			},
+			"view": {
+				Type:		schema.TypeString,
+				Description: "The DNS view name hosting the zone",
+				Computed:    true,
+			},
+			"class": {
+				Type:        schema.TypeString,
+				Description: "The class associated to the DNS zone.",
+				Computed:    true,
+			},
+			"class_parameters": {
+				Type:        schema.TypeMap,
+				Description: "The class parameters associated to IP space.",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func dataSourcednszoneRead(d *schema.ResourceData, meta interface{}) error {
+	s := meta.(*SOLIDserver)
+	d.SetId("")
+
+	// Building parameters
+	parameters := url.Values{}
+
+	parameters.Add("WHERE","dnszone_name='"+d.Get("name").(string)+"'" )
+	parameters.Add("limit","1")
+	parameters.Add("type",d.Get("type").(string))
+
+	// Sending the read request
+	resp, body, err := s.Request("get", "rest/dns_rr_list", &parameters)
```

Comment:
You are not calling the proper service, this one if for listing resource records, not zones.
You should call dns_zone_list.

Reply:
i just made the changed mentioned